### PR TITLE
net/tcp(unbuffered): removed excessive overwrites of conn->sndseq

### PR DESCRIPTION
**Pull request by a-lunev**
## Summary

conn->sndseq was updated in multiple places that was unreasonable and complicated.

## Impact

TCP

## Testing

### CODE REVIEW

File: net/tcp/tcp_input.c [hunk 1]
```diff
@@ -737,12 +736,14 @@ static void tcp_input(FAR struct net_driver_s *dev, uint8_t domain,
...
         }
 
+#ifdef CONFIG_NET_TCP_WRITE_BUFFERS
       /* Update sequence number to the unacknowledge sequence number.  If
        * there is still outstanding, unacknowledged data, then this will
        * be beyond ackseq.
        */
 
-      sndseq = tcp_getsequence(conn->sndseq);
+      uint32_t sndseq = tcp_getsequence(conn->sndseq);
```
Comment (xiaoxiang781216) [inline net/tcp/tcp_input.c:745]: need keep the line at line 687 since C89 doesn't support to define variables in the middle of code block.

Comment (a-lunev) [reply net/tcp/tcp_input.c]: Thank you. I returned the line to the place.

**Review by anchao — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by masayuki2009** [comment]
@a-lunev 

I noticed this PR has a side effect.
For example, if I try the wget command with spresnse:rndis + CONFIG_NET_TCP_WRITE_BUFFERS=n, it failed.
Actually, it works if I revert this PR. 

Please see the wireshark logs.

[spresense-rndis-wget-logs-20220102.zip](https://github.com/apache/incubator-nuttx/files/7798596/spresense-rndis-wget-logs-20220102.zip)

**Comment by a-lunev** [comment]
> @a-lunev
> 
> I noticed this PR has a side effect. For example, if I try the wget command with spresnse:rndis + CONFIG_NET_TCP_WRITE_BUFFERS=n, it failed. Actually, it works if I revert this PR.
> 
> Please see the wireshark logs.
> 
> [spresense-rndis-wget-logs-20220102.zip](https://github.com/apache/incubator-nuttx/files/7798596/spresense-rndis-wget-logs-20220102.zip)

Thank you for letting me know. I was able to reproduce the issue based on ftpc command. As I can see, tcp_sendfile is affected in unbuffered mode. I'm investigating the issue.

**Comment by a-lunev** [comment]
@masayuki2009 could you please review and test #5138 for the found issue?

**Pull request merged**